lib/pacemaker/pcmk_sched_promotable.c

```diff
@@ -644,12 +644,14 @@ promotion_attr_value(const pe_resource_t *rsc, const pe_node_t *node,
 {
     char *attr_name = NULL;
     const char *attr_value = NULL;
+    enum pe__rsc_node node_type = pe__rsc_node_assigned;
 
-    CRM_CHECK((rsc != NULL) && (node != NULL) && (name != NULL), return NULL);
-
+    if (pcmk_is_set(rsc->flags, pe_rsc_provisional)) {
```

Comment:
I wonder if `pe_node_attribute_calculated()` should do this on its own instead of taking an argument... which would eliminate the need for the `enum pe__rsc_node` type altogether (until a compatibility break, when we can use it to replace the magic int argument of `native_location()`). Although I guess we could pass an `enum pe__rsc_node` to `native_location()` even now as long as we don't change the signature, since the values are defined as `{0, 1, 2}`.

The only other caller of `pe_node_attribute_calculated()` is `get_node_score()`, which gets called only at unpack time.

I hadn't looked too closely at the callers this time around, just cleaned up my old commits.